src/FixedDataTable.js

```diff
@@ -1056,6 +1056,19 @@ var FixedDataTable = React.createClass({
 
     this._scrollHelper.setViewportHeight(bodyHeight);
```

Comment:
After some investigation, it seems the real bug is this line. We update the viewport height _after_ having calculated the rows. This makes our clamping operation invalid.

I think all we really need to do is check if scrollTop === scrollY.